### DISCUSSION TITLE
scikit-learn-intelex looks like it needs to mature

### DISCUSSION THREAD

**Discussion by celestinoxp** [pycaret/pycaret#2903] [General]
Does anyone use scikit-learn-intelex inside pycaret?

I tried testing and started having problems all over the place.

On github I saw a post about errors and that we should install the version from the source to have the most updated version with the errors corrected... after all the errors continue...

For starters, intelex's compatibility with scikit-learn isn't the best...

I use version 1.2dev, (ok is dev) I tried and gave errors..
However I installed the latest stable version scikit-learn 1.1.2, it gave problems too.
Then I tried the scikit-learn version 1.1.1 it also gave errors.
In a last attempt, I tested with scikit-learn 1.0.2 it worked, but.... it consumed all the ram memory 16gb, pycaret was unresponsive and I had to abort.

In short, it's too much trouble for me.

Has anyone tried it? If yes, does it work? If yes, with which versions?

**Comment by ngupta23** [comment]
I tried adding this for time series this weekend (https://github.com/pycaret/pycaret/pull/2898) and noticed some issues as well, e.g. the execution would get stuck when using cross-validation across multiple cores. But tests seem to be ok, so I have added it for now. Will check performance after it is merged. Any way, it is not default functionality and user needs to set this explicitly after installing scikit-learn-intelex so should not be a problem from a pycaret installation perspective. If the issue persists with it and it can not be resolved, we may just disable it in the future but for now, it is ok.

@PivovarA do you have any thoughts on this?

**Comment by PivovarA** [comment]
Hi Everyone!
At first I would like to clarify that the latest version of intelex - 2021.6.3 should work fine with the latest release of scikit-learn. We had issues with the compatibility, but everything should work fine now. I highly recommend installing scikit-learn-intelex via pip. We are working on improvements of compatibility.
As for running cross-validation on multiple cores, this is something we will need to check on our side as well. Thanks for pointing this out. Intelex uses all cores by default, so when you try to run several algorithms at once, you may encounter unpredictable behavior. We will think about how it can be improved on our side

**Reply by celestinoxp** [reply]
Hello,
@PivovarA  can you test again with scikit-learn 1.1.2? I installed scikit-lear-intelex the version from source, version: 2021.20220822.171739. and the errors I described above occurred. Could there be incompatibilities with other packages? Please test it yourself and if it works without errors, let us know which versions of the software you are using. Note: I tested on Windows 11, with python 3.9.12 and latest pycaret from source with all fixes.
Please paste the output of from `pycaret import show_versions; show_versions()`

**Reply by ngupta23** [reply]
> As for running cross-validation on multiple cores, this is something we will need to check on our side as well. Thanks for pointing this out. Intelex uses all cores by default, so when you try to run several algorithms at once, you may encounter unpredictable behavior. We will think about how it can be improved on our side

@PivovarA Since intelex is integrated with the regression and classification module for now, do you have any working example notebooks  for this? These modules use cross-validation by default across multiple cores. So I would think that this issue would show up here as well if the engine is changed to `intelex`. If you can share some working Google Colab notebook, that would be nice to have as reference. Thanks!

**Reply by ethanglaser** [reply]
Hello,
The intelex team has prepared a working [Google Colab notebook](https://colab.research.google.com/drive/16GvSREOIb2iI8GGMRoqKypsnFq6tDXJA?usp=sharing) demonstrating the runtime boost achieved when using the sklearnex engine in pycaret. Experiments were conducted on several pycaret-integrated datasets and algorithms compatible with intel's accelerations. The results are shown towards the end of the notebook.
